service/src/main/java/io/camunda/service/AdHocSubProcessActivityServices.java

```diff
@@ -38,15 +39,24 @@ public CompletableFuture<AdHocSubProcessInstructionRecord> activateActivities(
         new BrokerActivateAdHocSubProcessActivityRequest()
             .setAdHocSubProcessInstanceKey(request.adHocSubProcessInstanceKey());
 
-    request.elements().stream()
-        .map(AdHocSubProcessActivateActivityReference::elementId)
-        .forEach(brokerRequest::addElement);
+    request
+        .elements()
+        .forEach(
+            element ->
+                brokerRequest.addElement(
+                    element.elementId(), getDocumentOrEmpty(element.variables())));
+
+    brokerRequest.cancelRemainingInstances(
+        request.cancelRemainingInstances != null && request.cancelRemainingInstances());
 
     return sendBrokerRequest(brokerRequest);
   }
 
   public record AdHocSubProcessActivateActivitiesRequest(
-      String adHocSubProcessInstanceKey, List<AdHocSubProcessActivateActivityReference> elements) {
-    public record AdHocSubProcessActivateActivityReference(String elementId) {}
+      String adHocSubProcessInstanceKey,
+      List<AdHocSubProcessActivateActivityReference> elements,
+      Boolean cancelRemainingInstances) {
```

Comment:
```suggestion
      boolean cancelRemainingInstances) {
```

🔧 How about making this a primitive type? The `RequestMapper` can take care of mapping the missing value to a default false value, which is also what is described it'll be in the specification.

Reply:
Yes I agree ☝️Let's handle the `null->false` in RequestMapper to be backward compatible on API level only? Maybe we don't need to be backward compatible on the services layer?